src/analytics/analytics-tracker-window.ts

```diff
@@ -1,9 +1,60 @@
+import { isFunction } from 'lodash';
+
 export interface AnalyticsTracker {
     track(step: string, data: any): void;
-    hit(hitName: string, data: any): void;
-    hasPayloadLimit(data: any): boolean;
 }
 
 export default interface AnalyticsTrackerWindow extends Window {
     analytics: AnalyticsTracker;
+    ga(command: string, eventName: string, payload: any): void;
 }
+
+export const sendGoogleAnalytics = (type: string, payload: any) => {
+    const _window = window as unknown as AnalyticsTrackerWindow;
+
+    if (isFunction(_window?.ga)) {
+        _window.ga('send', type, {
+            ...payload,
+            nonInteraction: false,
+        });
+    }
+};
+
+export const isGoogleAnalyticsAvailable = () => {
+    const _window = window as unknown as AnalyticsTrackerWindow;
+
+    return isFunction(_window?.ga);
+};
+
+/**
+ * Max size of the payload for the Google Analytics module
+ * if the limit will be succeeded, the GA throwing a silent error,
+ * and only in debug mode you can see it
+ */
+export const ANALYTICS_MAX_URI_LENGTH = 8096;
+
+export const isPayloadSizeLimitReached = (obj: any) => {
```

Comment:
Do you have to use `any`?

Reply:
we can create AnalyticsPayload in analytics-step-tracker.ts and use instead of any

Reply:
changed to AnalyticsPayload type

Reply:
Changed to AnalyticPayload

Reply:
I tried to refactor this but finally decided that’s a wrong way and we can produce more bugs.

Reply:
Also we will continue working on this for Segment Beta.

Reply:
changed to AnalyticPayload interface